### ISSUE TITLE
azure function apps

### ISSUE THREAD

**Issue by Vanshata17** [marcduiker/azure-functions-university#101]
### 📚 Lesson Name

Http Trigger and Blob trigger

### 💬 Lesson Language

- [ ] .NET Core
- [X] .NET 5
- [ ] PowerShell
- [ ] TypeScript

### 📝 Question

Take the http trigger and retrieve the input(Student details) from the request body.

Input contains Student First Name, Last Name, Course, Passed Out year and Percentage.

Student Id is autoincrement Value and is the primary key for the table.

Validate the Student by passing first name and Last name to DB,

If the student  exists Update the Percentage and Passed Out year.

If the student not exists, Insert all the Details.

After successfully done the insert/update operations,  save the student details( Student Id, Student First Name, Last Name, Course, Passed Out year and Percentage) to the blob.

**Comment by lechnerc77** [comment]
Hi @Vanshata17 ,

welcome to Azure Functions University. I think I am not 100% clear about you question: currently there is no HTTÜP or Blob trigger lesson about .NET5 available in the lessons, do you reference the .NETCore lessons?

You question does not contain a question, but a description. Can you please rephrase it to a question?

Regards
Christian

**Comment by Vanshata17** [comment]
How do i pass multiple parameters in http trigger. I need to get first name
and last name from the query string in http trigger. How do i do that?

**Comment by lechnerc77** [comment]
From the caller perspective you concatenate them via `&`:   

```shell
http://localhost:7071/api/HttpTrigger1?firstname=Mark&lastname=Hamill
```

In your code you can then access them via the name you assigned in the call. Taking the example from above this would mean:

* TypeScript:

   ```typescript
   let firstname = req.query. firstname
   let lastname = req.query. lastname
   ````

*  .NET:

   ```csharp
   string firstname = req.Query["firstname"];
   string lastname = req.Query["lastname"];
   ```

**Comment by Vanshata17** [comment]
Ok so if i use postman and pass parameter(first name,last name) in the
query then how do i get that in json format in the body? And how can i code
the last part of http trigger default code, if there are multiple
parameters?
My task is basically using http trigger and retrieving the input from
request body and displaying it in json format. Could you please help me
with this? Input contains First name,last name,course,passed out year and
percentage.

**Comment by Vanshata17** [comment]
Even after passing the input in json format, it is asking for input in
query string

**Comment by lechnerc77** [comment]
Let us shortly summarize the requirement: 

- you want an HTTP triggered function
- the function should receive data via the request body
- the data should be taken from the request and returned to the caller

As you want to transfer the data via the request body, you are using a POST request that calls the Function endpoint with the data in the request body as JSON.  Due to the POST request you cannot use query parameters to transfer the data. 

Your scenario is covered in the HTTP trigger lesson i. e. [Changing the template for POST requests](https://github.com/marcduiker/azure-functions-university/blob/main/lessons/dotnetcore31/http/http-lesson-dotnet.md#3-changing-the-template-for-post-requests)

Concerning your error message, I would assume that there is a problem with an `IF ... ELSE` statement that always expects a query parameter from the caller. Please take a look at the code in the section mentioned above as this highlights how to deal with the different types of requests.  

In general, I would highly recommend to re-watch the HTTP trigger lesson on YouTube, as you scenario is discussed and demoed there.

**Comment by Vanshata17** [comment]
Thanks a lot for your reply. After retrieving the input from query i need
to pass  that to to database (sql server) to validate it. So how should i
proceed?

**Comment by Vanshata17** [comment]
This is the requirement basically here db is sql server

**Comment by lechnerc77** [comment]
@Vanshata17 ,

your questions  jumping between different problems and it seems to me that your questions do not touch any lessons or their respective content or the homework connected to it. 

In case you have generic questions around Azure Functions:
-  please refer to the official documentation: <https://docs.microsoft.com/azure/azure-functions/>
-  use one of the "usual "channels  to address  your questions like StackOverflow (<https://stackoverflow.com/questions/tagged/azure-functions>)

As issues in this GitHub repository should focus on the Azure Functions University and should not serve as a generic Q&A like e.g. StackOverflow  I close this issue
